examples/format_seqFISH.py

```diff
@@ -0,0 +1,197 @@
+        mappings.append({
+            Features.CODEWORD: [{
+                Indices.ROUND.value: r, Indices.CH.value: c - 1, Features.CODE_VALUE: 1
+            } for r, c in channel_series.items()],
+            Features.TARGET: gene
+        })
+
+    return {
+        "version": "0.0.0",
+        "mappings": mappings
+    }
+
+
+@click.command()
+@click.option("--input-dir", type=str, required=True, help="input directory containing images")
+@click.option("--output-dir", type=str, required=True, help="output directory for formatted data")
+@click.option("--codebook-csv", type=str, help="csv file containing barcode:target mapping")
+def cli(input_dir: str, output_dir: str, codebook_csv: str) -> int:
+    """CLI entrypoint for spaceTx format construction for SeqFISH data
+
+    Parameters
+    ----------
+    input_dir : str
+        directory containing input multi-page TIFF files for a single field of view, separated by
+        the imaging round they were acquired in and named <1-index round>.tif
+    output_dir : str
+        directory containing output files. Will be created if it does not exist.
+    codebook_csv : str
+        name of the codebook csv file containing barcode information for this field of view.
+
+    Notes
+    -----
+    - each round is organized as [z, ch, [x|y], [x|y]] -- the order of x and y are not known, but
+      since this script uses dummy coordinates, this distinction is not important.
+    - The spatial organization of the field of view is not known to the starfish developers,
+      so they are filled by dummy coordinates
+    - Raw data (input for this tool) for this experiment can be found at:
+      s3://spacetx.starfish.data.public/seqfish/
+    - Processed data (output of this tool) can be found at:
+      s3://spacetx.starfish.data.public/20181211/seqfish/ and accessed in `starfish.data.SeqFISH`
+
+    Returns
+    -------
+    int :
+        Returns 0 if successful
+    """
+    os.makedirs(output_dir, exist_ok=True)
+    primary_tile_fetcher = SeqFISHTileFetcher(os.path.expanduser(input_dir))
+
+    # This is hardcoded for this example data set
+    primary_image_dimensions: Mapping[Union[str, Indices], int] = {
+        Indices.ROUND: 5,
+        Indices.CH: 12,
+        Indices.Z: 29,
+    }
+
+    # def post_process_func(experiment_json_doc):
+    #     experiment_json_doc["codebook"] = "codebook.json"
+    #     return experiment_json_doc
+
+    write_experiment_json(
+        path=output_dir,
+        fov_count=1,
+        primary_image_dimensions=primary_image_dimensions,
+        aux_name_to_dimensions={},
+        primary_tile_fetcher=primary_tile_fetcher,
+        tile_format=ImageFormat.TIFF,
+        # postprocess_func=post_process_func,
+        dimension_order=(Indices.ROUND, Indices.CH, Indices.Z)
+    )
+
+    if codebook_csv:
+        with open(os.path.join(output_dir, "codebook.json"), "w") as f:
+            codebook = parse_codebook(codebook_csv)
```

Comment:
IMO, we should be constructing a Codebook object and using the code in that class to serialize.